rclcpp_components/src/component_manager.cpp

```diff
@@ -160,6 +159,16 @@ ComponentManager::create_node_options(const std::shared_ptr<LoadNode::Request> r
                 "Extra component argument 'use_intra_process_comms' must be a boolean");
       }
       options.use_intra_process_comms(extra_argument.get_value<bool>());
+    } else if (extra_argument.get_name() == "use_global_arguments") {
+      if (extra_argument.get_type() != rclcpp::ParameterType::PARAMETER_BOOL) {
+        throw ComponentManagerException(
+                "Extra component argument 'use_global_arguments' must be a boolean");
+      }
+      options.use_global_arguments(extra_argument.get_value<bool>());
+      RCLCPP_WARN(
+        get_logger(),
+        "This behavior is not recommended due to potential side-effect "
```

Comment:
A bit more detail / context here may be good.

Reply:
if the custom node constructor overwrites these options with default value (rejects specified options), these options cannot be changed or effective.

```
  SomeCustomNode(
    rclcpp::NodeOptions options
  )
  : Node(
      "custom_node",
      options.allow_undeclared_parameters(true).
      automatically_declare_parameters_from_overrides(true))
  {
  ...<snip>    
  }
```

i think that should be documented here? i might be missing something...

i mean if we can change the option for the node, that something we can do as user? and if that leads to side-effect or anything, that is the responsibility for node implementation, because it takes the arguments w/o check?

Reply:
maybe it can take some explanation from

https://github.com/ros2/rclcpp/blob/fa3a6fa597c5d40b2fce46e5fb95f541e62076e7/rclcpp/include/rclcpp/node_options.hpp#L152-L156

Reply:
Would `use_global_arguments is true by default in nodes, but is not recommended in a component manager. If true, this will cause this node's behavior to be influenced by global arguments, not only those targeted at this node.` work for you?

Reply:
> use_global_arguments is true by default in nodes, but is not recommended in a component manager. 

`use_global_arguments ` is true by default for `NodeOptions` ,but false in component manager currently, i think we should keep this default value `false` for component manager, what do you think about ? @SteveMacenski

Reply:
Yes, we are not changing the default. This warning is just telling people that `true` is default in general, but recommended not to be in this situation.